doc/develop/manifest/external/greybus.rst

```diff
@@ -0,0 +1,86 @@
+.. _external_module_greybus:
+
+Greybus
+#######
+
+Introduction
+************
+
+Greybus is a lightweight, message-based protocol framework that provides a
+standardized way for hosts to access hardware functions implemented on
+remote modules. It was originally developed for modular systems and
+defines a set of well-specified operation protocols for common peripheral
+classes such as GPIO, I²C, SPI, PWM, and firmware updates.
```

Comment:
Just for feedback: it helps to understand *over* which protocol this can work, as "can I use it with my boards" boils down to that (i.e. Ethernet-only? USB carrier? Possible to send Greybus messages over I2C?).

Easy confusion between the extra interfaces and protocols it can provide to the host and those it can use for supporting the physical/internal communication between the host and device.